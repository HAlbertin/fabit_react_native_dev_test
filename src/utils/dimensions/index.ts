import { Dimensions } from 'react-native';

const deviceWidth = Dimensions.get('window').width;
const deviceHeight = Dimensions.get('window').height;

const DimensionUtils = {
  deviceWidth,
  deviceHeight,
};

export default DimensionUtils;
