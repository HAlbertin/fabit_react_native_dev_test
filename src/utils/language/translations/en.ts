import { ITranslations } from './translations.interface';

const en: ITranslations = {
  /**
   * Languages
   */
  Language_Pt: 'Portuguese',
  Language_De: 'German',
  Language_En: 'English',
  Language_Es: 'Spanish',
  Language_It: 'Italian',
  Language_Fr: 'French',

  /**
   * LanguageScreen
   */
  LanguageScreen_SelectLanguage: 'Select your Language',
  LanguageScreen_ContinueButton: 'Continue',

  /**
   * LoginScreen
   */
  LoginScreen_HeaderTitle: 'Enter you email address',
  LoginScreen_Email: 'Email',
  LoginScreen_EmailPlaceholder: 'Your email address',
  LoginScreen_ContinueButton: 'Continue',
  LoginScreen_EmailInvalid: 'Invalid email',
  LoginScreen_EmailRequired: 'Email is required',
  LoginScreen_ErrorMessage_InvalidRequest: 'Something went wrong',
  LoginScreen_ErrorMessage_Unknow: 'Something went wrong',
  LoginScreen_ErrorMessage_EmailRequired: 'Email Required',
  LoginScreen_ErrorMessage_InvalidEmail: 'Invalid Email',
  LoginScreen_ErrorMessage_AccountSuspended: 'Account Suspended',
  LoginScreen_ErrorMessage_AccountClosed: 'Account Closed',
  LoginScreen_ErrorMessage_AccountLocked: 'Account locked',

  /**
   * SignupScreen
   */
  SignupScreen_HeaderTitle: 'Your new account',
  SignupScreen_Subtitle: 'Bacon ipsum dolor amet kielbasa filet mignon biltong hamburger tri-tip sirloin.',
  SignupScreen_CountryLabel: 'What country do you live in?',
  SignupScreen_CountryPlaceholder: 'Select country',
  SignupScreen_Bottomsheet_Country: 'Country',
  SignupScreen_StateLabel: 'Which state do you live in?',
  SignupScreen_StatePlaceholder: 'State',
  SignupScreen_Bottomsheet_StateLabel: 'State',
  SignupScreen_ErrorMessage_InvalidRequest: 'Something went wrong',
  SignupScreen_ErrorMessage_Unknow: 'Something went wrong',
  SignupScreen_ErrorMessage_EmailRequired: 'Email Required',
  SignupScreen_ErrorMessage_InvalidEmail: 'Invalid email',
  SignupScreen_ErrorMessage_AccountSuspended: 'Account suspended',
  SignupScreen_ErrorMessage_AccountClosed: 'Account Closed',
  SignupScreen_ErrorMessage_AccountLocked: 'Account Locked',
  SignupScreen_ErrorMessage_StateNotSupported: 'Invalid State',
};

export default en;
