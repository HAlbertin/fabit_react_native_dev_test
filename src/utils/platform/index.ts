import { Platform } from 'react-native';

const isAndroid = () => Platform.OS === 'android';
const isIos = () => Platform.OS === 'ios';

const IOS = '3';
const ANDROID = '2';
const getPlatform = () => (PlatformUtils.isIos ? IOS : ANDROID);

const PlatformUtils = {
  isAndroid,
  isIos,
  getPlatform,
};

export default PlatformUtils;
