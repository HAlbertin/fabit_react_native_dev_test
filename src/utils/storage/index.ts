import { MMKV } from 'react-native-mmkv';
import StorageConstants from '../../constants/storage.constants';

const storage = new MMKV();

type storageKeys = keyof typeof StorageConstants;

const setItem = (keyName: storageKeys, value: string | number | boolean) => {
  storage.set(StorageConstants[keyName], value);
};

const getItem = (keyName: storageKeys) => storage.getString(StorageConstants[keyName]);

const deleteItem = (keyName: storageKeys) => storage.delete(keyName);

const StorageUtils = {
  setItem,
  getItem,
  deleteItem,
};

export default StorageUtils;
