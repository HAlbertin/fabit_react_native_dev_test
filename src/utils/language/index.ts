import * as Localization from 'expo-localization';
import i18n from 'i18n-js';
import StorageUtils from '../storage';
import de from './translations/de';
import en from './translations/en';
import es from './translations/es';
import fr from './translations/fr';
import it from './translations/it';
import pt from './translations/pt';
import { TranslationKeys } from './translations/translations.interface';

export const getCurrentLocale = () => Localization.locale;

export const DEFAULT_LANGUAGE = 'en';
export const DEFAULT_TRANSLATION = {
  key: DEFAULT_LANGUAGE,
  value: 'English',
};

export const getCurrentLanguage = () => StorageUtils.getItem('USER_LANGUAGE') ?? DEFAULT_LANGUAGE;

export const changeLanguage = (newLocale: string) => {
  i18n.locale = newLocale;
};

export const translate = (key: TranslationKeys): string => i18n.t(key);

export const LANGUAGES = [
  { lang: 'pt', file: pt, name: 'Language_Pt' },
  { lang: 'en', file: en, name: 'Language_En' },
  { lang: 'es', file: es, name: 'Language_Es' },
  { lang: 'it', file: it, name: 'Language_It' },
  { lang: 'de', file: de, name: 'Language_De' },
  { lang: 'fr', file: fr, name: 'Language_Fr' },
];

const setupInitialLanguage = () => {
  i18n.fallbacks = true;

  // Set the key-value pairs for the different languages you want to support.
  i18n.translations = LANGUAGES.reduce((translations, { lang, file }) => {
    translations[lang] = file;
    return translations;
  }, {});

  // Set the locale once at the beginning of your app.
  i18n.locale = getCurrentLocale();
};

setupInitialLanguage();
