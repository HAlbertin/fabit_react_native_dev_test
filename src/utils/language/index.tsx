import * as Localization from 'expo-localization';
import i18n from 'i18n-js';
import de from './translations/de';
import en from './translations/en';
import es from './translations/es';
import fr from './translations/fr';
import it from './translations/it';
import pt from './translations/pt';
import { TranslationKeys } from './translations/translations.interface';

const LANGUAGES = [
  { lang: 'pt', file: pt },
  { lang: 'en', file: en },
  { lang: 'es', file: es },
  { lang: 'it', file: it },
  { lang: 'de', file: de },
  { lang: 'fr', file: fr },
];

// Set the key-value pairs for the different languages you want to support.
i18n.translations = LANGUAGES.reduce((translations, { lang, file }) => {
  translations[lang] = file;
  return translations;
}, {});

// Set the locale once at the beginning of your app.
i18n.locale = Localization.locale;

export const changeLanguage = (newLocale: string) => {
  i18n.locale = newLocale;
};

export const translate = (key: TranslationKeys): string => i18n.t(key);
