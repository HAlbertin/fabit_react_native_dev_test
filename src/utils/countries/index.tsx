import countries from 'i18n-iso-countries';
import de from 'i18n-iso-countries/langs/de.json';
import en from 'i18n-iso-countries/langs/en.json';
import es from 'i18n-iso-countries/langs/es.json';
import fr from 'i18n-iso-countries/langs/fr.json';
import it from 'i18n-iso-countries/langs/it.json';
import pt from 'i18n-iso-countries/langs/pt.json';
import Flag from '../../components/flag';
import { getCurrentLanguage } from '../language';

const __setConfig = () => {
  countries.registerLocale(en);
  countries.registerLocale(es);
  countries.registerLocale(pt);
  countries.registerLocale(it);
  countries.registerLocale(fr);
  countries.registerLocale(de);
};

const GetAllCountries = () => {
  const countriesObj = countries.getNames(getCurrentLanguage(), { select: 'official' });
  const countriesList = Object.entries(countriesObj).map(e => ({
    key: e[0],
    value: e[1],
    icon: <Flag isoCode={e[0]} />,
  }));
  return countriesList;
};

__setConfig();

const CountriesUtils = {
  GetAllCountries,
};

export default CountriesUtils;
