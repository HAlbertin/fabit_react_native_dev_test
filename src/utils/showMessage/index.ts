import { Alert } from 'react-native';
import { translate } from '../language';
import { TranslationKeys } from '../language/translations/translations.interface';

const showMessage = (title: TranslationKeys, message?: TranslationKeys) =>
  Alert.alert(translate(title), message ? translate(message) : '');

const ShowMessageUtils = {
  showMessage,
};

export default ShowMessageUtils;
