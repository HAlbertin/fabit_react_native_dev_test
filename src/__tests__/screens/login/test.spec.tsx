import { act, fireEvent } from '@testing-library/react-native';
import LoginScreen from '../../../screens/login';
import en from '../../../utils/language/translations/en';
import { renderWithNavigator } from '../../../utils/test';

describe('<LoginScreen />', () => {
  const INVALID_EMAIL = 'aaaaa';
  const EMPTY_EMAIL = '';
  const VALID_EMAIL = 'teste@teste.com';

  it('should match snapshot', () => {
    const { toJSON } = renderWithNavigator(LoginScreen);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should show email errors for input', async () => {
    const { getByTestId, getByText } = renderWithNavigator(LoginScreen);

    const emailInput = getByTestId('email-input');

    await act(async () => {
      await fireEvent.changeText(emailInput, INVALID_EMAIL);
    });

    const invalidLabel = getByText(en.LoginScreen_EmailInvalid);
    expect(invalidLabel).toBeTruthy();

    await act(async () => {
      await fireEvent.changeText(emailInput, EMPTY_EMAIL);
    });

    const requiredLabel = getByText(en.LoginScreen_EmailRequired);
    expect(requiredLabel).toBeTruthy();
  });

  it('should enable submit button', async () => {
    const { getByTestId, queryByTestId } = renderWithNavigator(LoginScreen);

    const emailInput = getByTestId('email-input');

    await act(async () => {
      await fireEvent.changeText(emailInput, VALID_EMAIL);
    });

    const invalidLabel = queryByTestId(en.LoginScreen_EmailInvalid);
    const requiredLabel = queryByTestId(en.LoginScreen_EmailRequired);

    expect(invalidLabel).toBeFalsy();
    expect(requiredLabel).toBeFalsy();
  });
});
