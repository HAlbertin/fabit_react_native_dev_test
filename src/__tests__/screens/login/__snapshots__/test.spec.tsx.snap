// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen /> should match snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="Enter you email address"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RNCSafeAreaView
            edges={
              Array [
                "right",
                "left",
                "bottom",
              ]
            }
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <View
              onLayout={[Function]}
              style={
                Array [
                  Array [
                    Object {
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                  ],
                  Object {
                    "paddingBottom": 0,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "justifyContent": "space-between",
                      "marginBottom": 32,
                      "marginTop": 32,
                      "paddingBottom": 0,
                      "paddingLeft": 24,
                      "paddingRight": 24,
                      "paddingTop": 0,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "flex-start",
                        "flexDirection": "column",
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <Text
                    color="#000"
                    style={
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 16,
                        },
                      ]
                    }
                    testID="email-input-label"
                  >
                    Email
                  </Text>
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "rgb(231, 234, 231)",
                          "borderTopLeftRadius": 4,
                          "borderTopRightRadius": 4,
                        },
                        Object {
                          "backgroundColor": "#FFFFFF",
                          "borderColor": "#000",
                          "borderRadius": 4,
                          "borderStyle": "solid",
                          "borderWidth": 1,
                          "marginTop": 8,
                          "width": "100%",
                        },
                      ]
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "bottom": 0,
                          "height": 2,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "transform": Array [
                            Object {
                              "scaleY": 0.5,
                            },
                          ],
                          "zIndex": 1,
                        }
                      }
                    />
                    <View
                      style={
                        Array [
                          Object {
                            "paddingBottom": 0,
                            "paddingTop": 0,
                          },
                          Object {
                            "minHeight": 62,
                          },
                        ]
                      }
                    >
                      <TextInput
                        autoCapitalize="none"
                        autoCorrect={false}
                        blurOnSubmit={false}
                        editable={true}
                        keyboardType="email-address"
                        maxFontSizeMultiplier={1.5}
                        multiline={false}
                        onBlur={[Function]}
                        onChangeText={[Function]}
                        onFocus={[Function]}
                        placeholder="Your email address"
                        placeholderTextColor="#707070"
                        selectionColor="#FFB81C"
                        style={
                          Array [
                            Object {
                              "flexGrow": 1,
                              "margin": 0,
                            },
                            Object {
                              "paddingLeft": 12,
                              "paddingRight": 12,
                            },
                            Object {
                              "height": 62,
                            },
                            Object {
                              "paddingBottom": 0,
                              "paddingTop": 0,
                            },
                            Object {
                              "color": "#000",
                              "fontFamily": "System",
                              "fontSize": 16,
                              "fontWeight": undefined,
                              "textAlign": "left",
                              "textAlignVertical": "center",
                            },
                            false,
                            Array [
                              Object {},
                            ],
                          ]
                        }
                        testID="email-input"
                        textContentType="emailAddress"
                        underlineColorAndroid="transparent"
                        value=""
                      />
                    </View>
                  </View>
                  <Text
                    color="#E02020"
                    style={
                      Array [
                        Object {
                          "color": "#E02020",
                          "fontSize": 16,
                        },
                      ]
                    }
                    testID="email-input-error"
                  >
                    
                  </Text>
                </View>
                <View
                  collapsable={false}
                  style={
                    Object {
                      "backgroundColor": "rgba(0, 0, 0, 0.12)",
                      "borderColor": "transparent",
                      "borderRadius": 0,
                      "borderStyle": "solid",
                      "borderWidth": 0,
                      "elevation": 0,
                      "minWidth": 64,
                      "shadowColor": "#000000",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      Object {
                        "disabled": true,
                      }
                    }
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Object {
                          "borderRadius": 4,
                        },
                      ]
                    }
                    testID="button-contained"
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "center",
                          },
                          Object {
                            "backgroundColor": "#999999",
                            "height": 64,
                          },
                        ]
                      }
                    >
                      <Text
                        numberOfLines={1}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#000",
                              "fontFamily": "System",
                              "fontSize": 16,
                              "fontWeight": "400",
                            },
                            Object {
                              "textAlign": "left",
                            },
                            Array [
                              Object {
                                "letterSpacing": 1,
                                "marginHorizontal": 16,
                                "marginVertical": 9,
                                "textAlign": "center",
                              },
                              undefined,
                              false,
                              Object {
                                "color": "rgba(0, 0, 0, 0.32)",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": "500",
                              },
                              Object {
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": "500",
                              },
                              Object {
                                "color": "#101820",
                                "fontWeight": "bold",
                              },
                            ],
                          ]
                        }
                      >
                        <Text
                          color="#101820"
                          style={
                            Array [
                              Object {
                                "color": "#101820",
                                "fontSize": 16,
                              },
                            ]
                          }
                          testID="button-contained-text"
                        >
                          Continue
                        </Text>
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RNCSafeAreaView>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </View>
</View>
`;
